cyclonedx/model/component.py

```diff
@@ -1457,6 +1458,11 @@ def cpe(self) -> Optional[str]:
 
     @cpe.setter
     def cpe(self, cpe: Optional[str]) -> None:
+        if cpe:
+            try:
+                CPE(cpe)
+            except NotImplementedError:
+                raise ValueError(f'Invalid CPE format: {cpe}')
```

Comment:
This behavioral change is considered a breaking change.
Not a blocker, but a justification.